src/components/FeatureViews/AdminPortal/PanelManager.js

```diff
@@ -0,0 +1,386 @@
+            "update": updateFunction
+        }
+    ];
+    Possible Field Types: 
+        text, active, toggle, daterange, integer, money, enumText, enumCollection
+        ...
+        for enumCollection, the object needs "fetch" and "update" attributes 
+
+    @prop {function} fetchFunction array of functions for fetching records.
+    ex. api.fetchCategories
+    
+    @prop {function} statusFunction function which returns fetch? status
+    ex. useSelector(({RequestStatus}) => RequestStatus.category)
+
+    @prop {function} updateFunction update 
+
+    @prop {function} selectorHook selector that gets records from store
+
+    @prop {array} recordArray
+ *    
+ **/
+    const zip = rows => rows[0].map((_, c) => rows.map(row=>row[c]));
+
+    //recordState
+    const [recordList, setRecordList] = useState([]);
+    const [directoryData, setDirectoryData] = useState({
+        "ids": [],
+        "editing" : []
+    });
+
+    const records = props.collectionData
+
+    const dispatch = useDispatch();
+
+    const api = useMemo(
+        () => bindActionCreators(adminActions, dispatch),
+    [dispatch]
+    ); 
+
+    const defaults = {
+        "editable": "false",
+        "align": "left",
+        "type": "text"
+    };
+
+    const addDefaults = (fields) => {
+        let fieldsWithDefaults = {};
+        Object.keys(fields).forEach((field) => {
+            if (field !== "ids" && field !== "editing") {
+                Object.keys(defaults).forEach((key) => {
+                    if(fields[field].hasOwnProperty(key)) {
+                        //Do nothing, property is already set
+                    } else {
+                        fields[field][key] = defaults[key];
+                    }
+                })
+            }
+
+        })
+        
+        return fields
+    };
+
+    // useEffect(() => {
+    //     console.log("records: " + records)
+    //     if(records.length !== recordList.length) {
+    //         let parsedRecordList = records.map((record) => ({
+    //             ...record,
+    //             editing: false,
+    //         }));
+    //         setRecordList(parsedRecordList);
+    //     }
+    // }, [records]);
+
+    const fieldsWithDefaults = addDefaults(props.fields);
+
+
+
+    useEffect(() => {
+        const constructDirectoryObject = (fieldInfo, recordArray) => {
+            fieldInfo["ids"] = [];
+            fieldInfo["editing"] = [];
+            recordArray.forEach((record, index) => {
+                Object.keys(record).forEach((key) => {
+                    if (key !== "id") {
+                        if(fieldInfo[key]["values"]){
+                            fieldInfo[key]["values"].push(record[key])
+                        } else {
+                        fieldInfo[key]["values"] = [record[key]]
+                        }
+                    } else if (key === "id") {
+                        fieldInfo["ids"].push(record[key])
+                    }
+                })
```

Comment:
What does `recordArray` look like after this line? Does it actually reset all editing values to `false`?